File: LinkedList /getKthFromLast.py
'''
	{
		# Node Class
		class Node:
		    def __init__(self, data):   # data -> value stored in node
		        self.data = data
		        self.next = None
	}
'''
class Solution:
    def getKthFromLast(self, head, k):
        #code here
        slow = head
        fast = head
    
        
        for _ in range(k):
            if not fast:
                return -1  
            fast = fast.next
    
        
        while fast:
            slow = slow.next
            fast = fast.next
    
        
        return slow.data if slow else -1

File: LinkedList /getKthFromLast.java
/* Structure of node
class Node
{
    int data;
    Node next;
    Node(int d) {data = d; next = null; }
} */

class Solution {

    // Function to find the data of kth node from
    // the end of a linked list.
    int getKthFromLast(Node head, int k) {
        // Your code here
        Node first = head;
        Node second = head;

        // Move first pointer k steps ahead
        for (int i = 0; i < k; i++) {
            if (first == null) {
                return -1; // k is more than the number of nodes
            }
            first = first.next;
        }

        // Move both pointers until first reaches the end
        while (first != null) {
            first = first.next;
            second = second.next;
        }

        return (second != null) ? second.data : -1;
    }
}
